File: CPP/NikhilSinghh/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int water=0, max_water=0;
        int s=0, e=h.size()-1;
        while(s<e){
            int dist=e-s;
             int min_height = min(h[e], h[s]);
                     water = dist * min_height;
                  max_water = max(water, max_water);
            if(h[s]<h[e])
                s++;
            else
                e--;
        }
        return max_water;
    }
};

File: Java/Akash-212/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        //  int maxArea = 0;

        //  for (int i = 0; i < height.length - 1; i++) {
        //      for (int j = i + 1; j < height.length; j++) {
        //          int minHeight = Math.min(height[i], height[j]);
        //          int width = j - i;
        //          int area = minHeight * width;
        //          maxArea = Math.max(maxArea, area);
        //      }
        //  }

        //  return maxArea;
        int maxArea = 0;
        int left = 0;
        int right = height.length - 1;

        while (left < right) {
            int minHeight = Math.min(height[left], height[right]);
            int width = right - left;
            int area = minHeight * width;
            maxArea = Math.max(maxArea, area);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }
}
